lua/avante/file_selector.lua

```diff
@@ -158,6 +158,19 @@ function FileSelector:fzf_ui(handler)
   }))
 end
 
+---@type FileSelectorHandler
+function FileSelector:snacks_picker_ui(handler)
+  Snacks.picker.files({
+    exclude = self.selected_filepaths,
+    confirm = function(picker)
+      picker:close()
+      local items = picker:selected({ fallback = true })
+      local files = vim.tbl_map(function(item) return item.file end, items)
+      handler(#files == 1 and files[1] or files)
```

Comment:
Maybe this will work too:

```suggestion
      handler(files)
```